mongo/client_encryption.go

```diff
@@ -85,6 +87,10 @@ func (ce *ClientEncryption) CreateEncryptedCollection(ctx context.Context,
 		return nil, nil, errors.New("nil CreateCollectionOptions")
 	}
 
+	if ce.closed {
```

Comment:
Could we move this above the opts check? Knowing the client is closed is more important than if the options have been set, as setting the options would still result in an invalid usage of this method.